src/System.Net.Http/tests/StressTests/HttpStress/ClientOperations.cs

```diff
@@ -324,6 +324,22 @@ public static class ClientOperations
                         ValidateContent(content, await m.Content.ReadAsStringAsync());
                     }
                 }),
+                
+                ("POST Duplex Dispose",
+                async ctx =>
+                {
+                    // try to reproduce conditions described in https://github.com/dotnet/corefx/issues/39819
+                    string content = ctx.GetRandomString(0, ctx.MaxContentLength);
+                    Version httpVersion = ctx.GetRandomHttpVersion();
+
+                    using (var req = new HttpRequestMessage(HttpMethod.Post, "/duplex") { Version = httpVersion, Content = new StringDuplexContent(content) })
+                    using (HttpResponseMessage m = await ctx.SendAsync(req, HttpCompletionOption.ResponseHeadersRead))
+                    {
+                        ValidateHttpVersion(m, httpVersion);
+                        ValidateStatusCode(m);
+                        m.Content.Dispose();
```

Comment:
HttpResponeMessage.Dispose will dispose of its Content, so this line isn't necessary.  The difference between this and "POST Duplex" then is whether we read the response body or not.  I'd have expected then that we would have seen similar issues arise from cancellation occurring at appx the same time while running "POST Duplex".  Does that make sense, and if so, do you know why we weren't seeing such failures there (or were we)?

Reply:
I hadn't stumbled upon any failures after running the test suite for about an hour last week (admittedly that was using the default cancellation rate of 10% spread distributed over all operations). On the other hand, running just this operation with a cancellation rate of zero yielded consistently yielded failures almost immediately. I would like to see if this similar errors can be observed with high cancellation rates using just "Post Duplex Slow".

Reply:
Running either POST Duplex or POST Duplex Slow with 100% cancellation did not yield any errors, whereas the new operation does reproduce errors.

Reply:
Ok. That makes me question how much variation we're actually getting from the cancellation support.

Reply:
I suppose it might be because cancellation is triggered almost immediately after a request task has been created, but currently do not force any other causal relationship. But even so I would expect us to hit outliers, given sufficient runs.

Reply:
FYI, the issue requires that request body completion (NOT cancellation) and response body dispose happen at approximately the same time. That's what causes the race.

I'm not sure when StringDuplexContent will complete the request body, but since we did see failures with this variation, it must at least occasionally be completing it at approx. the same time as disposing the response.